lib/api/messages.js

```diff
@@ -3878,3 +3889,25 @@ function parseAddresses(data) {
     walk([].concat(data || []));
     return Array.from(addresses);
 }
+
+function getAttachmentCharset(mimeTree, attachmentId) {
+    if (mimeTree.attachmentId && mimeTree.attachmentId === attachmentId) {
+        // current mimeTree (sub)object has the attachmentId field, and it is the one we search
+        // get the parsedHeader -> content-type -> params -> charset
+
+        return [mimeTree.parsedHeader['content-type'].params.charset || 'UTF-8', true];
```

Comment:
There is near-zero chance that the `content-type` header is missing, so maybe use optional chaining to prevent exceptions

```js
mimeTree.parsedHeader['content-type']?.params?.charset
```